koku/masu/external/downloader/azure/azure_service.py

```diff
@@ -158,7 +163,19 @@ def get_file_for_key(self, key: str, container_name: str) -> BlobProperties:
         return report
 
     def get_latest_cost_export_for_path(self, report_path: str, container_name: str) -> BlobProperties:
-        return self._get_latest_blob_for_path(report_path, container_name, AzureBlobExtension.csv.value)
+        """
+        Get the latest cost export for a given path and container. Supports both CSV and GZIP formats.
+        """
+        blob = self._get_latest_blob_for_path(report_path, container_name, AzureBlobExtension.gzip.value)
```

Comment:
Lets not be guessing formats here. We should be looking this up and dynamically passing it everywhere its needed really. Use `describe_cost_management_exports` to lookup what the compression type is.

Inside this method we build the report definitions:
```report_def = {
                        "name": report.name,
                        "container": report.delivery_info.destination.container,
                        "directory": report.delivery_info.destination.root_folder_path,
                    }
```

We should just add compression to that list. Then we can set `self.compression = compression` inside `AzureReportDownloader`